packages/astro/src/runtime/server/render/astro/instance.ts

```diff
@@ -37,6 +37,7 @@ export class AstroComponentInstance {
 	}
 
 	async init(result: SSRResult) {
+		if (this.returnValue !== undefined) return this.returnValue;
```

Comment:
This part makes sure `this.returnValue` is cached. Otherwise we're calling `render()` then `init()`, which causes double head rendering.